File: wlr/SPX.Util/StringStorage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace SPX.Util
{
    public class StringStorage
    {
        private static readonly Dictionary<int, string> __slots = new Dictionary<int, string>();

        public static string Get(float value)
        {
            if (__slots.Keys.Count() == 0)
            {
                for (float ii = 0; ii < 1000; ii++)
                {
                    __slots.Add((int)ii, ii.ToString());
                }
            }
            if (__slots.ContainsKey((int)value))
            {
                return __slots[(int)value];
            }
            return value.ToString();
        }
    }
}

File: proj/src/spx/util/StringStorage.java
package spx.util;

import java.util.HashMap;

public class StringStorage {
	private static HashMap<Integer, String> __slots = new HashMap<Integer, String>();

	public static String Get(float value) {
		if (__slots.keySet().size() == 0) {
			for (int ii = 0; ii < 1000; ii++) {
				__slots.put((int) ii, Integer.toString(ii));
			}
		}
		if (__slots.containsKey((int) value)) {
			return __slots.get((int) value);
		}
		return Float.toString(value);
	}
}